pkg/deployments/tasks/20221101-l2-gauge-checkpointer/test/task.fork.ts

```diff
@@ -0,0 +1,234 @@
+    const gaugeControllerTask = new Task('20220325-gauge-controller', TaskMode.READ_ONLY, getForkedNetwork(hre));
+    const gaugeController = await gaugeControllerTask.instanceAt(
+      'GaugeController',
+      '0xC128468b7Ce63eA702C1f104D55A2566b13D3ABD' // gaugeControllerTask.output({ network: 'mainnet' }).GaugeController
+    );
+
+    const getGaugesData = async (gaugeInputs: [string, number][]) => {
+      return Promise.all(
+        gaugeInputs.map(async (gaugeInput) => {
+          return {
+            address: gaugeInput[0],
+            weight: await gaugeController['gauge_relative_weight(address)'](gaugeInput[0]),
+            expectedCheckpoints: gaugeInput[1],
+          };
+        })
+      );
+    };
+    const polygonRootGaugesData: GaugeData[] = await getGaugesData(polygonRootGauges);
+    const arbitrumRootGaugesData: GaugeData[] = await getGaugesData(arbitrumRootGauges);
+
+    // There are no optimism gauges that are correctly added to the Gauge Adder and the Gauge Controller at this point.
+    gauges.set(GaugeType.Polygon, polygonRootGaugesData);
+    gauges.set(GaugeType.Arbitrum, arbitrumRootGaugesData);
+  });
+
+  before('add gauges to checkpointer', async () => {
+    await Promise.all(
+      Array.from(gauges).map(([gaugeType, gaugesData]) => {
+        L2GaugeCheckpointer.addGauges(
+          gaugeType,
+          gaugesData.map((gaugeData) => gaugeData.address)
+        );
+      })
+    );
+  });
+
+  before('grant checkpoint permission to fees gauge checkpointer', async () => {
+    // Any gauge works; we just need the interface.
+    const gauge = await task.instanceAt('IStakelessGauge', gauges.get(GaugeType.Polygon)![0].address);
+    const govMultisig = await impersonate(GOV_MULTISIG, fp(100));
+
+    await authorizer
+      .connect(govMultisig)
+      .grantRole(
+        await authorizerAdaptor.getActionId(gauge.interface.getSighash('checkpoint')),
+        L2GaugeCheckpointer.address
+      );
+  });
+
+  it('checks that gauges were added correctly', async () => {
+    for (const [gaugeType, gaugesData] of gauges.entries()) {
+      expect(await L2GaugeCheckpointer.getTotalGauges(gaugeType)).to.be.eq(gaugesData.length);
+    }
+  });
+
+  describe('getTotalBridgeCost', () => {
+    context('when threshold is 1', () => {
+      itChecksTotalBridgeCost(fp(1));
+    });
+
+    context('when threshold is 0.0001', () => {
+      itChecksTotalBridgeCost(fp(0.0001));
+    });
+
+    context('when threshold is 0', () => {
+      itChecksTotalBridgeCost(fp(0));
+    });
+  });
+
+  describe('checkpoint', () => {
+    const checkpointedGauges: GaugeData[] = [];
+    let gaugeDataAboveMinWeight: GaugeData[] = [];
+    let minRelativeWeight: BigNumber;
+
+    // Gauges won't be checkpointed twice, so when the threshold is lowered and more gauges get above the threshold
+    // we need to filter out those that have already been checkpointed.
```

Comment:
Isn't it easier to use snapshots? The test setup otherwise becomes quite fragile

Reply:
I found it a bit weird to do so back then in a fork test, but I think it makes sense.

I've added an empty `sharedBeforeEach` that takes the snapshot and it does its job; no need to keep track of the checkpointed gauges now.